assets/js/admin/course-edit.js

```diff
@@ -1,36 +1,59 @@
-/**
- * It enables or disables the legacy meta boxes.
- *
- * @param {boolean} enable Whether enable or disable.
- */
-window.sensei_toggleLegacyMetaboxes = ( enable ) => {
-	// Side-effect - Prevent submit modules.
-	document
-		.querySelectorAll( '#module_course_mb input' )
-		.forEach( ( input ) => {
-			input.disabled = ! enable;
-		} );
-
-	const editPostSelector = wp.data.select( 'core/edit-post' );
-	const editPostDispatcher = wp.data.dispatch( 'core/edit-post' );
+import { select, dispatch } from '@wordpress/data';
 
-	const legacyMetaboxes = [
-		'meta-box-course-lessons',
-		'meta-box-module_course_mb',
-		'meta-box-course-video',
+( () => {
+	const COURSE_OUTLINE_NAME = 'sensei-lms/course-outline';
+	const OTHER_COURSE_BLOCKS = [
+		'sensei-lms/button-take-course',
+		'sensei-lms/button-contact-teacher',
+		'sensei-lms/course-progress',
 	];
 
-	legacyMetaboxes.forEach( ( legacyMetabox ) => {
-		if (
-			enable !== editPostSelector.isEditorPanelEnabled( legacyMetabox )
-		) {
-			editPostDispatcher.toggleEditorPanelEnabled( legacyMetabox );
-		}
-	} );
-};
+	const blockEditorSelector = select( 'core/block-editor' );
+	const editPostSelector = select( 'core/edit-post' );
+	const editPostDispatcher = dispatch( 'core/edit-post' );
+
+	/**
+	 * Toggle meta boxes depending on the blocks.
+	 */
+	window.sensei_toggleLegacyMetaboxes = () => {
+		if ( ! blockEditorSelector ) return;
+
+		const outlineBlockCount = blockEditorSelector.getGlobalBlockCount(
+			COURSE_OUTLINE_NAME
+		);
+		const otherCourseBlocksCount = OTHER_COURSE_BLOCKS.reduce(
+			( sum, blockName ) =>
+				sum + blockEditorSelector.getGlobalBlockCount( blockName ),
+			0
+		);
+
+		const legacyMetaboxes = [
+			{ name: 'meta-box-course-lessons', deps: outlineBlockCount },
```

Comment:
I think that this is something that should be added as an argument so each block can define its own metaboxes. Otherwise whenever we add a new block we need to update this function too.

Reply:
Hum, yeah! That can be a good optimization! Added it here: https://github.com/Automattic/sensei/pull/3818/commits/97f1f2da0f70b0a3020863d54525db7fdcc013b4

Reply:
I thought a little more about this and how we could refactor that. To implement something like that, probably we'd need to save something in the store or create a context for that, because every time we add/remove a component, we need to control if we still have some component handling the same metaboxes. So I believe it would be much more complex, and considering it's probably something temporary (I hope) until we remove the metaboxes, I think we could keep all this logic isolated and simple. WDYT?

Reply:
Cool let's keep it like this and revise the approach in case that we need something similar in another page.